community/server/src/main/java/org/neo4j/server/rrd/sampler/DatabasePrimitivesSampleableBase.java

```diff
@@ -27,23 +27,48 @@
 
 public abstract class DatabasePrimitivesSampleableBase implements Sampleable
 {
+    public interface SampleableDatabase
+    {
+        boolean isAvailable();
+    }
+
     private final NeoStoreProvider neoStore;
+    private final SampleableDatabase samapleableDatabase;
+    private double lastReadValue = 0d;
 
-    public DatabasePrimitivesSampleableBase( NeoStoreProvider neoStore )
+    public DatabasePrimitivesSampleableBase( NeoStoreProvider neoStore, SampleableDatabase sampleableDatabase )
     {
-        if( neoStore == null )
+        if ( neoStore == null )
         {
             throw new RuntimeException( "Database sampler needs a NeoStore to work, was given null." );
         }
         this.neoStore = neoStore;
-
+        this.samapleableDatabase = sampleableDatabase;
     }
 
-    protected NeoStore getNeoStore()
+    @Override
+    public double getValue()
     {
-        return neoStore.evaluate();
+        if ( samapleableDatabase.isAvailable() )
+        {
+            try
+            {
+                lastReadValue = readValue( neoStore.evaluate() );
+            }
+            catch ( Exception e )
+            {
+                /*
+                 * oh well, this is unfortunate, perhaps the db went down after the check, so let's ignore this problem
```

Comment:
the _not_ logging is on purpose, we don't want to be notified about this, we simply don't care